ui/app/pages/home/home.component.js

```diff
@@ -81,11 +83,18 @@ export default class Home extends PureComponent {
 
   componentDidUpdate () {
     const {
-      threeBoxSynced,
+      isNotification,
       setupThreeBox,
       showRestorePrompt,
       threeBoxLastUpdated,
+      threeBoxSynced,
+      totalUnapprovedCount,
     } = this.props
+
+    if (isNotification && totalUnapprovedCount === 0) {
+      global.platform.closeCurrentWindow()
```

Comment:
`componentDidMount` seems like a better place for this. `componentDidUpdate` isn't called on the initial render, so it could hang on the Home screen for a while in theory.

Reply:
Should we do both? I'm wondering whether it's possible for the component to mount with a positive number of pending confirmations that then go to 0 while the component is already mounted, thereby never closing the UI.

Reply:
Hmm... I was thinking that if it had a positive number of confirmations, it wouldn't get to this point in the first place (it'd be routed to another page). But that's not entirely true - it does route to the Home page for... probably just a single render, while the redirect happens. So yeah, both might make sense.